pkg/api/types.go

```diff
@@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
 	// Items is a list of ResourceQuota objects
 	Items []ResourceQuota `json:"items"`
 }
+
+// Secret holds secret data of a certain type
+type Secret struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	Data map[string][]byte `json:"data,omitempty"`
+	Type SecretType        `json:"type,omitempty"`
+}
+
+type SecretType string
+
+const (
+	SecretTypeOpaque SecretType = "opaque" // Default; arbitrary user-defined data
```

Comment:
@bgrant0607 @thockin will collect these i to a PR soon (might not be until
Monday).
On Sat, Feb 21, 2015 at 1:33 AM Brian Grant <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4514#discussion_r25120223>
> :
>
> > @@ -1309,3 +1317,27 @@ type ResourceQuotaList struct {
> >  	// Items is a list of ResourceQuota objects
> >  	Items []ResourceQuota `json:"items"`
> >  }
> > +
> > +// Secret holds secret data of a certain type
> > +type Secret struct {
> > +	TypeMeta   `json:",inline"`
> > +	ObjectMeta `json:"metadata,omitempty"`
> > +
> > +	Data map[string][]byte `json:"data,omitempty"`
> > +	Type SecretType        `json:"type,omitempty"`
> > +}
> > +
> > +type SecretType string
> > +
> > +const (
> > +	SecretTypeOpaque SecretType = "opaque" // Default; arbitrary user-defined data
>
> +1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/4514/files#r25120223>
> .
>